src/squidpy/tl/_sliding_window.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+import math
+from functools import reduce
+from typing import TYPE_CHECKING, Any, Literal, NamedTuple, Optional, Union
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from scanpy import logging as logg
+
+from squidpy._docs import d
+from squidpy._utils import NDArrayA
+from squidpy.gr._utils import _save_data
+
+__all__ = ["sliding_window"]
+
+
+@d.dedent
+def sliding_window(
+    adata: AnnData,
+    library_key: str | None = None,
+    window_size: int = 2500,
+    overlap: int = 0,
+    coord_columns: tuple[str, str] = ("globalX", "globalY"),
+    sliding_window_key: str = "sliding_window",
+    spatial_key: str = "spatial",
+    copy: bool = False,
+) -> AnnData:
```

Comment:
You annotate that you return an AnnData, but you either modify in place (None) or return a pandas df